data-integration/README.md

```diff
@@ -0,0 +1,341 @@
+        "phone": "8770586755",
+        "email": "johndoe@gmail.com",
+        "department": "IT"
+   }
+   ```
+
+2. Send a `POST` request to the service to create a task for the employee.
+
+    ```json
+    {
+        "taskId": 1001,
+        "description": "Analyze network performance",
+        "status": "IN_PROGRESS",
+        "employeeId": 1
+    }
+    ```
+
+3. Send a `PUT` request to the service to update the task status.
+   
+    ```json
+    {
+        "description": "Analyze network performance",
+        "status": "COMPLETED",
+        "employeeId": 1
+    }
+    ```
+
+4. Delete the task by giving the task ID of the completed task that needs to be deleted.
+
+### Concepts covered
+
+- REST API
+- The Ballerina persistence layer
+- CRUD operations
+
+## Develop the application
+
+### Step 1: Set up the workspace
+
+Install [Ballerina Swan Lake](https://ballerina.io/downloads/) and the [Ballerina Swan Lake extension](https://marketplace.visualstudio.com/items?itemName=wso2.ballerina) on VS Code.
+
+### Step 2: Develop the service
+
+Follow the instructions given in this section to develop the service.
+
+1. Create a new Ballerina project using the `bal` command and open it in VS Code.
+
+    ```bash
+    $ bal new data-integration
+    ```
+
+2. Use `bal persist` in the project root to initialize the persistence layer for the project. Specify the module name as `store` and the datastore as `mysql`. This will create a directory named `persist` in the project root directory. The directory will contain the `model.bal` file, which is used to define the data model.
+
+    ```bash
+    $ bal persist init --module store --datastore mysql
+    ```
+
+3. Open the `model.bal` file in the persist directory and define the data model as follows.
+
+    ```ballerina
+    import ballerina/persist as _;
+
+    type Employee record {|
+        readonly int id;
+        string name;
+        int age;
+        string phone;
+        string email;
+        string department;
+        Task[] tasks;
+    |};
+
+    type Task record {|
+        readonly int taskId;
+        string taskName;
+        string description;
+        TaskStatus status;
+        Employee employee;
+    |};
+
+    public enum TaskStatus {
+        NOT_STARTED,
+        IN_PROGRESS,
+        COMPLETED
+    };
+    ```
+
+    - The above data model defines a one to many relationship, where each `Employee` can have multiple assigned `Task`s, and each `Task` is associated with only one `Employee`.   
+
+    **Note:** 
+    > The entities defined in the `model.bal` file should be based on the [`persist model specification`](https://ballerina.io/learn/persist-model/).
+
+4. Generate the client objects, types, and scripts for the model using the following command. This will use the model defined in the `persist/model.bal` file to generate the persistence constructs in a module named `store` in a directory named `generated`.
+   
+    ```bash
+    $ bal persist generate --module store
```

Comment:
```suggestion
    $ bal persist generate
```
We can avoid the `--module store` part in this command as it is already executed in the init command above.